test/y2storage/storage_features_list_test.rb

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env rspec
+#
+# encoding: utf-8
+
+# Copyright (c) [2017,2019-2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "spec_helper"
+require "y2storage/storage_features_list"
+
+describe Y2Storage::StorageFeaturesList do
+  describe ".new" do
+    context "if no bit-field is specified" do
+      it "returns all possible features" do
+        expect(described_class.new).to contain_exactly(*Y2Storage::StorageFeature.all)
+      end
+    end
+
+    context "if the bit-field is zero" do
+      let(:bits) { 0 }
+
+      it "returns an empty list" do
+        expect(described_class.new(bits)).to be_empty
+      end
+
+      # The combination of this test and the corresponding unit test for
+      # StorageFeature#in_bitfield? ensures yast-storage-ng does not contain any
+      # feature that is not defined in libstorage-ng (a NameError exception
+      # would be raised).
+      it "calls #in_bitfield? for all registered features" do
```

Comment:
np: This description basically reveals an implementation detail.  Let's keep it as it is because, at least, we have a nice text documenting the reason.

Reply:
Thinking it better, I could move the code raising the `NameError` exception directly to the constructor of `StorageFeature`. There is no need to defer the check until `#in_bitfield?` is called. Failing right away when the object is created would be ok and everything (including the tests) would look way more natural.

Reply:
Done in a later commit already added to this PR.